contracts/bonding/BondingManager.sol

```diff
@@ -860,7 +865,9 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
     function updateDelegatorWithTokenPoolsShares(address _delegator, uint256 _endRound) internal {
         Delegator storage del = delegators[_delegator];
 
-        if (del.lastClaimTokenPoolsSharesRound > 0) {
+        // Only will have earnings to claim if you have a delegate
+        // If not delegated, skip the earnings claim process
+        if (del.delegateAddress != address(0)) {
```

Comment:
This check makes sure that a delegator does not try to claim earnings since its last claim round if it does not currently have a delegate because it could not possibly have any earnings to claim in this case. We do not check if the delegate is also a registered transcoder here even though the delegate would have to be a registered transcoder in order for there to be claimable earnings - this is to account for the case in which a transcoder gets slashed, but its delegators still need to claim earnings from past rounds while they were delegated to the now slashed transcoder